src/Services/DriverOptionsFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php declare(strict_types=1);
+
+namespace Facile\MongoDbBundle\Services;
+
+/**
+ * Class DriverOptionsFactory.
+ * @internal
+ */
+final class DriverOptionsFactory implements DriverOptionsInterface
```

Comment:
What I meant here https://github.com/facile-it/mongodb-bundle/pull/87#discussion_r325661644 
was adapting the configuration to accept a service reference (defined in your application) which is constraint to implement an interface (defined in this bundle) for buildind driver options.